### PR TITLE
Improve and expand on documentation for the API rate limiter

### PR DESCRIPTION
**Pull request by christarazi**
See commit msgs.

I would appreciate feedback on the wording and the readability of the docs
specifically. This PR's goal is to share and ease the understanding of the API
rate limiter.

### CODE REVIEW

File: Documentation/configuration/api-rate-limiting.rst [hunk 1]
```diff
@@ -68,17 +68,17 @@ Configuration Key                 Example   Default   Description
 ================================= ========= ========= =====================================================================================
 ``rate-limit``                    ``5/m``   None      Allowed requests per time unit in the format ``<number>/<duration>``.
 ``rate-burst``                    ``4``     None      Burst of API requests allowed by rate limiter.
-``min-wait-duration``             ``10ms``  0         Minimum wait duration each API call has to wait before being processed.
-``max-wait-duration``             ``15s``   0         Maximum duration an API call is allowed to wait before it fails.
-``estimated-processing-duration`` ``100ms`` 0         Estimated processing duration of an average API call. Used for automatic adjustment.
+``min-wait-duration``             ``10ms``  ``0```    Minimum wait duration each API call has to wait before being processed.
```
Comment (christarazi) [inline Documentation/configuration/api-rate-limiting.rst]: Extra backtick

**Review by tgraf — APPROVED** [review]
LGTM except for the extra tick and build failure

File: Documentation/configuration/api-rate-limiting.rst [hunk 2]
```diff
@@ -147,3 +147,90 @@ All API calls subject to rate limiting will expose :ref:`metrics_api_rate_limiti
...
+response.
+
+This is a common message when the rate limiter is doing its job of preventing
+too many parallel requests at once.
+
+.. code::
+
+   "Not processing API request. Wait duration exceeds maximum"
+
+The request has denied by the rate limiter because the request's waiting
```
Comment (qmonnet) [inline Documentation/configuration/api-rate-limiting.rst]:
```suggestion
The request has been denied by the rate limiter because the request's waiting
```

**Review by qmonnet — COMMENTED** [review]
Would that make sense to add a comment for `wait()` in `pkg/rate/api_limiter.go` to mention that the error messages are documented and that any update or addition should also be reported to `Documentation/configuration/api-rate-limiting.rst`?

**Comment by christarazi** [comment]
Thanks @qmonnet, that's a good idea. I've added a comment on `wait()` to reference the docs.

**Comment by christarazi** [comment]
Marking for backport to 1.7, 1.8, and 1.9 as those versions contain the API rate limiter, and would benefit from this change.

**Pull request merged**